0015-automatic-cluster-upgrades/0015-automatic-cluster-upgrades.md

```diff
@@ -0,0 +1,95 @@
+# RFC 0015 - Automatic Cluster Upgrades
+
+## User stories
+
+- As a user, and also as KaaS team provider, I would like to have the clusters running in the latest version automatically, and not worry about CVEs or EOL of the cluster components.
+
+- As a user, I would like to define maintenance windows to ensure clusters are not upgraded outside of these hours.
+
+- As a user I would like to mark a cluster as frozen and block upgrades for certain period of time.
+
+- As a user I want to define a policy that enforces some environments are upgraded before than others to an agreed version.
+
+- As a user I want to schedule a upgrade for a certain day/time.
+
+- As a user, and also as KaaS team provider, I want to capture the result of upgrade.
+
+## Reasoning
+
+Our customers need to plan every upgrade on their side and manage the upgrades for every GS version. Every customer has their own requirement when they need to schedule upgrades (freeze periods, maintenance windows,...). At the same time, Cloud Native projects and Giant Swarm are releasing more versions every time. It provokes a lot of time preparing and executing these upgrades which is not usually given much value to any of the stakeholders. 
+
+Giant Swarm designed a versioning schema (semver), which includes all components on the stack, to ensure idempotency and expose the level of impact that changes bring to the customer. Patches and Minor releases should have no impact on the customer workloads, which mean can be applied in the environments at any time. 
```

Comment:
at any time? do you mean "automatically" instead? "any time" sounds like it wouldn't respect the maintenance / freeze windows.

Reply:
well I can rephrase it but I wanted to emphasize `patches` and `minors` should be harmless. I mention the use case of customer need to define constraints later, is that enough?